sync/src/chain.rs

```diff
@@ -437,6 +448,10 @@ impl ChainSync {
 		trace!(target: "sync", "{} -> NewBlock ({})", peer_id, h);
 		let header: BlockHeader = try!(header_rlp.as_val());
  		let mut unknown = false;
+		{
+			let peer = self.peers.get_mut(&peer_id).expect("ChainSync: unknown peer");
```

Comment:
this `expect` doesn't look like it's a fundamental logic error. if it is, add a comment explaining why it can never ever happen and other pre-condition `assert!`s in the code to make the proof clear.